File: ncarg2d/man/gks/gssgt.m
.\"
.\"	$Id: gssgt.m,v 1.16 2008-12-23 00:03:03 haley Exp $
.\"
.TH GSSGT 3NCARG "March 1993" UNIX "NCAR GRAPHICS"
.SH NAME
GSSGT (Set segment transformation) - Associates a segment transformation
with a named segment.
.SH SYNOPSIS
CALL GSSGT(SGNA,M)
.SH C-BINDING SYNOPSIS
#include <ncarg/gks.h>
.sp
void gset_seg_tran(Gint seg_name, const Gtran_matrix tran_matrix);
.SH DESCRIPTION
.IP SGNA 12
(Integer, Input) - A segment name as used in a previous GCRSG call.
.IP M 12
(Real, Input) - A 2x3 array giving the GKS segment transformation to
be associated with the segment specified in SGNA.
.SH USAGE
When the segment named in SGNA is displayed, the coordinates of its
primitives are transformed by the matrix specified in M.
.sp
.SH ACCESS
To use GKS routines, load the NCAR GKS-0A library 
ncarg_gks.
.SH SEE ALSO
Online: 
gactm, gevtm, gclsg, gcrsg, gcsgwk, gdsg, gqopsg, gqsgus., gset_seg_tran
.sp
Hardcopy: 
"User's Guide for NCAR GKS-0A Graphics"
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br
The use of this Software is governed by a License Agreement.

File: ncarg2d/src/libncarg_gks/awi/gssgt.f
C
C	$Id: gssgt.f,v 1.4 2008-07-27 00:21:02 haley Exp $
C                                                                      
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      SUBROUTINE GSSGT(SGNA,M)
C
C  SET SEGMENT TRANSFORMATION
C
      INTEGER ESSGT
      PARAMETER (ESSGT=64)
C
      include 'gkscom.h'
C
      INTEGER SGNA
      REAL    M(2,3)
C
C  Check if GKS is in the proper state.
C
      CALL GZCKST(7,ESSGT,IER)
      IF (IER .NE. 0) RETURN
C
C  Check that the segment name is valid.
C
      IF (SGNA.LT.0 .OR. SGNA.GT.99) THEN
        ERS = 1
        CALL GERHND(120,ESSGT,ERF)
        ERS = 0
        RETURN
      ENDIF
C
C  Check if the segment exists.
C
      NXST = 0
      DO 200 I=1,NUMSEG
        IF (SEGS(I) .EQ. SGNA) THEN
          NXST = 1
          ISNDX = I
        ENDIF
  200 CONTINUE
      IF (NXST .EQ. 0) THEN
        ERS = 1
        CALL GERHND(122,ESSGT,ERF)
        ERS = 0
        RETURN
      ENDIF
C
C  Define the segment transformation matrix for segment SGNA.
C
      SEGT(ISNDX,1,1) = M(1,1)
      SEGT(ISNDX,1,2) = M(1,2)
      SEGT(ISNDX,1,3) = M(1,3)
      SEGT(ISNDX,2,1) = M(2,1)
      SEGT(ISNDX,2,2) = M(2,2)
      SEGT(ISNDX,2,3) = M(2,3)
C
      RETURN
      END
